src/Altinn.App.Core/Implementation/AppResourcesSI.cs

```
@@ -104,66 +109,49 @@ public byte[] GetText(string org, string app, string textResource)
         /// <inheritdoc />
         public Application GetApplication()
         {
-            // Cache application metadata
-            if (_application != null)
-            {
-                return _application;
-            }
-
-            string filedata = string.Empty;
-            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.ApplicationMetadataFileName;
             try
             {
-                if (File.Exists(filename))
+                var task = Task.Run(async () => await _appMetadata.GetApplicationMetadata());
+                task.Wait();
+                if (task.IsCompletedSuccessfully)
                 {
-                    filedata = File.ReadAllText(filename, Encoding.UTF8);
+                    return task.Result;
                 }
-
-                _application = JsonConvert.DeserializeObject<Application>(filedata)!;
-                return _application;
             }
             catch (Exception ex)
             {
-                _logger.LogError("Something went wrong when fetching application metadata. {0}", ex);
-                return null;
+                _logger.LogError(ex, "Something went wrong fetching application metadata");
+                return null;   
```

Let it crash and burn?

---

Crash and burn implemented in latest version :)